pallets/portfolio/src/benchmarking.rs

```diff
@@ -0,0 +1,115 @@
+            // However, without this hack, B will be the first char after a bump in number of chars.
+            // i.e. the sequence will go A,B...Z,BA,BB...ZZ,BAA. We want the sequence to start with A.
+            // Subtracting 1 here means we are doing 1 indexing rather than 0.
+            // i.e. A = 1, B = 2 instead of A = 0, B = 1
+            calc_base26((n / 26) - 1, base_26);
+        }
+        let character = n % 26 + 65;
+        base_26.push(character as u8);
+    }
+    let mut base_26 = Vec::new();
+    calc_base26(n, &mut base_26);
+    Ticker::try_from(base_26.as_slice()).unwrap()
+}
+
+benchmarks! {
+    _ {}
+
+    create_portfolio {
+        // Length of portfolio name
+        let i in 1 .. 500;
+
+        let (_, target_origin, target_did) = make_account::<T>("target", 0);
+        let portfolio_name = PortfolioName(vec![65u8; i as usize]);
+        let next_portfolio_num = NextPortfolioNumber::get(&target_did);
+    }: _(target_origin, portfolio_name.clone())
+    verify {
+        assert_eq!(Portfolios::get(&target_did, &next_portfolio_num), portfolio_name);
+    }
+
+    delete_portfolio {
+        let (_, target_origin, target_did) = make_account::<T>("target", 0);
+        let portfolio_name = PortfolioName(vec![65u8; 5]);
+        let next_portfolio_num = NextPortfolioNumber::get(&target_did);
+        Module::<T>::create_portfolio(target_origin.clone().into(), portfolio_name.clone())?;
+        assert_eq!(Portfolios::get(&target_did, &next_portfolio_num), portfolio_name);
+    }: _(target_origin, next_portfolio_num.clone())
+    verify {
+        assert!(!Portfolios::contains_key(&target_did, &next_portfolio_num));
+    }
+
+    move_portfolio_funds {
+        // Number of assets being moved
+        let i in 1 .. 500;
+        let mut items = Vec::with_capacity(i as usize);
+        let (_, target_origin, target_did) = make_account::<T>("target", 0);
+        let first_ticker = generate_ticker(0u64);
+        let amount = T::Balance::from(10);
+        let portfolio_name = PortfolioName(vec![65u8; 5]);
+        let next_portfolio_num = NextPortfolioNumber::get(&target_did);
+        let default_portfolio = PortfolioId::default_portfolio(target_did);
+        let user_portfolio = PortfolioId::user_portfolio(target_did, next_portfolio_num.clone());
+
+        for x in 0..i as u64 {
+            let ticker = generate_ticker(x);
+            items.push(MovePortfolioItem {
+                ticker,
+                amount: amount,
+            });
+            <PortfolioAssetBalances<T>>::insert(&default_portfolio, &ticker, amount);
+        }
+
+        Module::<T>::create_portfolio(target_origin.clone().into(), portfolio_name.clone())?;
+
+        assert_eq!(<PortfolioAssetBalances<T>>::get(&default_portfolio, &first_ticker), amount);
+        assert_eq!(<PortfolioAssetBalances<T>>::get(&user_portfolio, &first_ticker), 0.into());
```

Comment:
For my education, shouldn't this be in the verify step?

Reply:
In the highlighted lines, we are verifying that the environment for the benchmark is set up properly. The benchmark hasn't been executed yet.

The format of a benchmark unit is something like this

```
$benchmark_name {
    // Stuff that should be done before the actual benchmark.
    // This sets up the environment required for the benchmark.
    // Any asserts here verify that the environment is setup correctly.
    // These are executed before the benchmark extrinsic is executed. 
}: $extrinsic_to_benchmark($parameters_to_pass_to_extrinsic) // `_` can be used instead of `$extrinsic_to_benchmark` is `$extrinsic_to_benchmark` and `$benchmark_name` are the same
verify {
    // Stuff to verify that the extrinsic/benchmark was executed properly. 
    // This verifies that the benchmark did the changes that were expected.
    // These are executed after the benchmark extrinsic is executed.
}

Reply:
Ah; thanks for clarifying!